wrappers/src/app_cs.hpp

```diff
@@ -133,6 +133,22 @@ namespace binding {
     extern std::function<BsonCallbackT> s_bson_callback;
     extern std::function<ApiKeysCallbackT> s_api_keys_callback;
 
+    inline auto get_string_callback_handler(void* tcs_ptr) {
+        return [tcs_ptr](const std::string* response, util::Optional<AppError> err) {
```

Comment:
`response` is a `const std::string&` in the signature of `call_function`. Here you're passing it as a pointer. It'd be better you stick to have it as a reference for coherence with the original signature.
Additionally, the check at line 146 is unnecessary as response can never be null, given that it's a reference and not a pointer.

Reply:
![image](https://user-images.githubusercontent.com/44238293/205940201-a28a8fa6-afe8-496b-b8b8-cff413f9f9b0.png)

I think I'm looking at the correct overload and this one takes a `const std::string*` (on core version 13.1.0), are you looking at a different version? It won't build if I change it to const std::string& on your suggested line :/

Reply:
Never mind. They changed the signature later on. If you look at the [PR](https://github.com/realm/realm-core/pull/5984/files) linked in the original issue you'll see that the parameter has a `const std::string&`
![image](https://user-images.githubusercontent.com/4258222/205942291-cbdbe250-d35b-42b8-85ce-08e92bf565b4.png)

But if you look at the [current signature](https://github.com/realm/realm-core/blob/89be4acd41391ce14313ca9d29f4a010696b12d9/src/realm/object-store/sync/app_service_client.hpp#L51), it's `const std::string*`.
I should have checked the current signature and not the PR one. You'd have spotted it right away by compiling the wrappers locally. I did not test before commenting.

Reply:
No worries, would have made sense if it was the case! :D I'll revert